### PR TITLE
 Update the cartographer convention build to v0.1.0-build.5 pre release

### PR DESCRIPTION
**Pull request by kkavitha**

### CODE REVIEW

**Comment by kkavitha** [comment]
cc: @scothis  @cirocosta

File: src/cartographer/config/overlays/sa-arn-annotation-overlay.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Copyright 2021 VMware
```
Comment (scothis) [inline src/cartographer/config/overlays/sa-arn-annotation-overlay.yaml:1]:
```suggestion
# Copyright 2022 VMware
```

File: src/cartographer/config/overlays/sa-arn-annotation-overlay.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#@ load("@ytt:overlay", "overlay")
+#@ load("@ytt:data", "data")
+
+#@ if/end hasattr(data.values, "arn_role_id"):
+#@overlay/match by=overlay.subset({"apiVersion":"v1","kind":"ServiceAccount","metadata":{"namespace":"cartographer-system", "name":"cartographer-conventions-controller-manager"}}), expects="1+"
```
Comment (scothis) [inline src/cartographer/config/overlays/sa-arn-annotation-overlay.yaml:19]: there can only be one

```suggestion
#@overlay/match by=overlay.subset({"apiVersion":"v1","kind":"ServiceAccount","metadata":{"namespace":"cartographer-system", "name":"cartographer-conventions-controller-manager"}}), expects="1"
```

File: src/cartographer/config/objects/ytt/schema.yaml [hunk 3]
```diff
@@ -17,3 +17,4 @@
 ca_cert_data: ""
 excluded_components:
 - ""
+arn_role_id: ""
```
[the remark this answers is not in the document]
Comment (cirocosta) [reply src/cartographer/config/objects/ytt/schema.yaml:20]: addressed 😁 

generation of the package doing it as expected

```yaml
apiVersion: data.packaging.carvel.dev/v1alpha1
kind: Package
metadata:
  name: cartographer.tanzu.vmware.com.0.0.0
spec:
  refName: cartographer.tanzu.vmware.com
  version: 0.0.0
#...
        arn_role_id:
          type: string
          description: 'Optional: ID of the arn role that has access to pull images from ECR container registry'
          default: ""
# ...
```

**Review by cirocosta — APPROVED** [review]
lgtm!

```console
$ tanzu package install carto --package-name cartographer.tanzu.vmware.com --version 0.0.0 --values-file ./values.yaml
 Installing package 'cartographer.tanzu.vmware.com'
 Getting package metadata for 'cartographer.tanzu.vmware.com'
 Creating service account 'carto-default-sa'
 Creating cluster admin role 'carto-default-cluster-role'
 Creating cluster role binding 'carto-default-cluster-rolebinding'
 Creating secret 'carto-default-values'
 Creating package resource
 Waiting for 'PackageInstall' reconciliation for 'carto'
 'PackageInstall' resource install status: Reconciling
 'PackageInstall' resource install status: ReconcileSucceeded

 Added installed package 'carto'
```

with `values.yaml` being

```yaml
arn_role_id: foo
```

we indeed get

```console
$ k -n cartographer-system get sa cartographer-conventions-controller-manager -o json | jq '.metadata.annotations'
{
  "eks.amazonaws.com/role-arn": "arn:aws:iam::foo:role/conventions-service",
  ...
}
```

**Comment by kkavitha** [comment]
The input should be `arn:aws:iam::2048996713745:role/conventions-service` instead of just the number

**Comment by kkavitha** [comment]
Marking this as a draft as the key name for the arn is being discussed.

**Comment by kkavitha** [comment]
Installed the test package like above, seeing this:
```
kubectl -n cartographer-system get sa cartographer-conventions-controller-manager -o json | jq '.metadata.annotations'
{
  "eks.amazonaws.com/role-arn": "arn:aws:iam::foo:role/conventions-service",
  ...
}
```

**Review by scothis — APPROVED** [review]

**Review by cirocosta — APPROVED** [review]

**Pull request merged**